File: Assets/Scripts/Network/ClientPackets/MessagePacket.cs
using System;
using System.Text;

public class MessagePacket : ClientPacket {
    private string _text;
    private string _sender;

    public MessagePacket(string text) : base(0x02) {
        SetText(text);
        BuildPacket(Encoding.ASCII.GetBytes(text));
    }

    public void SetSender(string sender) {
        _sender = sender;
    }

    public void SetText(string text) {
        _text = text;
    }
}

File: java-server/src/main/java/com/shnok/serverpackets/MessagePacket.java
package com.shnok.serverpackets;

public class MessagePacket extends ServerPacket {
    private String _text;
    private String _sender;

    public MessagePacket(String sender, String text) {
        super((byte)0x02);

        setText(text);
        setSender(sender);
        buildMessageData();
    }

    public void setSender(String sender) {
        _sender = sender;
    }

    public void setText(String text) {
        _text = text;
    }

    public void buildMessageData() {
        writeS(_sender);
        writeS(_text);
        buildPacket();
    }
}
